backend/main.py

```
@@ -828,6 +830,92 @@ async def complete_session(project_id: str, session_id: str):
+            "insights_found": consolidation_result.total_insights_processed + consolidation_result.total_insights_skipped,
+            "session_relevance": consolidation_result.session_relevance
+        }
+        
+        logger.info(f"Session end completed successfully: {consolidation_result.status}")
+        return response_data
+        
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.error(f"Error ending session with consolidation: {e}")
+        raise HTTPException(status_code=500, detail=f"Session end failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)